### PR TITLE
KAFKA-14239: Merge StateRestorationIntegrationTest into RestoreIntegrationTest

### PR DESCRIPTION
**Pull request by ahmedsobeh**
This PR makes the following changes:

- Moves the only test in StateRestorationIntegrationTest into RestoreIntegrationTest
- Deletes StateRestorationIntegrationTest

### CODE REVIEW

**Comment by guozhangwang** [comment]
Thanks @ahmedsobeh ! Will take a look.

**Comment by ahmedsobeh** [comment]
> Thanks @ahmedsobeh ! Will take a look.

had a small import issue that I just fixed, should be good now.

**Comment by guozhangwang** [comment]
There's still jenkins failures: 

```
[2022-09-23T00:16:58.037Z] [ant:checkstyle] [ERROR] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-12670/streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java:161:22: Name 'APPLICATION_ID' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalFinalVariableName]

[2022-09-23T00:16:58.037Z] [ant:checkstyle] [ERROR] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-12670/streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java:162:22: Name 'STATE_STORE_NAME' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalFinalVariableName]

[2022-09-23T00:16:58.037Z] [ant:checkstyle] [ERROR] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-12670/streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java:163:22: Name 'INPUT_TOPIC' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalFinalVariableName]

[2022-09-23T00:16:58.037Z] [ant:checkstyle] [ERROR] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-12670/streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java:164:22: Name 'OUTPUT_TOPIC' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalFinalVariableName]

[2022-09-23T00:16:58.037Z] [ant:checkstyle] [ERROR] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-12670/streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java:168:20: Variable 'streamsConfiguration' should be declared final. [FinalLocalVariable]
```

File: streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java [hunk 1]
```diff
@@ -156,14 +228,16 @@ public void shouldRestoreStateFromSourceTopic(final boolean stateUpdaterEnabled)
         final Properties props = props(stateUpdaterEnabled);
         props.put(StreamsConfig.TOPOLOGY_OPTIMIZATION_CONFIG, StreamsConfig.OPTIMIZE);
 
-        // restoring from 1000 to 4000 (committed), and then process from 4000 to 5000 on each of the two partitions
+        // restoring from 1000 to 4000 (committed), and then process from 4000 to 5000
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java]: The newlines introduced below till line 496 seems to be auto from IDE, could you revert them if they are not intentional?

File: streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java [hunk 2]
```diff
@@ -97,11 +102,13 @@ public class RestoreIntegrationTest {
 
     public static final EmbeddedKafkaCluster CLUSTER = new EmbeddedKafkaCluster(NUM_BROKERS);
 
+
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java]: Are these new lines intentional? If not could you revert them?

Comment (ahmedsobeh) [reply streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java]: Used a formatter that seems to have added a couple of few lines. Removed 👍

**Review by guozhangwang — APPROVED** [review]
LGTM!

**Pull request merged**